config/routes.rb

```diff
@@ -66,6 +65,8 @@
           get 'studies/:accession/download', to: 'site#download_data', as: :site_study_download_data
           get 'studies/:accession/stream', to: 'site#stream_data', as: :site_study_stream_data
 
+          post 'studies/:accession/user_annotation', to: 'user_annotations#create_user_annotation', as: :site_study_create_user_annotation
```

Comment:
I'd like to stick to REST resource conventions as much as possible, especially within the studies namespace.  I would move this up to the 'studies' resource, that will let you define it with something like
`resources :user_annotations, only: [:create]`. 
This you can rename your api controller method to just `create` and also eliminate the need to modify study_aware, since the study id will get passed the same way as the other newer API calls.  

Might make sense to have a quick call on this -- I know rails routing suggestions are never as simple to follow as they are to make :)

Reply:
Thanks for the discussing this through with me.  It's done in the latest code.